src/Psalm/Internal/Analyzer/CommentAnalyzer.php

```diff
@@ -1254,4 +1258,59 @@ public static function splitDocLine($return_block)
 
         return [$type];
     }
+
+    /**
+     * @param ParsedDocblock $parsed_docblock
+     *
+     * @return void
+     *
+     * @throws DocblockParseException
+     */
+    private static function checkDuplicatedTags(ParsedDocblock $parsed_docblock)
+    {
+        if (count($parsed_docblock->tags['return'] ?? []) > 1
+            || count($parsed_docblock->tags['psalm-return'] ?? []) > 1
+            || count($parsed_docblock->tags['phpstan-return'] ?? []) > 1
+        ) {
+            throw new DocblockParseException('Found duplicated @return or prefixed @return tag');
+        }
+
+        self::checkDuplicatedParams($parsed_docblock->tags['param'] ?? []);
+        self::checkDuplicatedParams($parsed_docblock->tags['psalm-param'] ?? []);
+        self::checkDuplicatedParams($parsed_docblock->tags['phpstan-param'] ?? []);
+    }
+
+    /**
+     * @param array<int, string> $param
+     *
+     * @return void
+     *
+     * @throws DocblockParseException  if an invalid string or a duplicate is found
+     */
+    private static function checkDuplicatedParams(array $param)
+    {
+        $splitted_params = self::splitAllDocLine($param);
+
+        $param_names = array_column($splitted_params, 1);
+        if (count($param_names) !== count(array_unique($param_names))) {
+            throw new DocblockParseException('Found duplicated @param or prefixed @param tag');
+        }
+    }
+
+    /**
+     * @param array<int, string> $lines
+     *
+     * @throws DocblockParseException  if an invalid string is found
+     *
+     * @return array<int, array<string>>
+     */
+    public static function splitAllDocLine(array $lines)
+    {
+        $splitted_lines = [];
+        foreach ($lines as $offset => $line) {
+            $splitted_lines[$offset] = self::splitDocLine($line);
```

Comment:
This the only quibble I have – `splitDocLine` does a lot we don't need – a simple `explode(' ', $line)` should be sufficient (and much faster).

```suggestion
            $splitted_lines[$offset] = \explode(' ', $line);
```